### DISCUSSION TITLE
Live Whisper transcription, and Voice Assistant!

### DISCUSSION THREAD

**Discussion by Nikorasu** [openai/whisper#465] [Show and tell]
This is my *"real-time"* implementation of Whisper, which sentence-by-sentence transcribes microphone audio to the terminal.
[https://github.com/Nikorasu/LiveWhisper](https://github.com/Nikorasu/LiveWhisper)
Uses the sounddevice library, because my system has conflicts with with PyAudio..

I've also included my Voice Assistant project, which uses this Whisper implementation as a replacement for the standard speech-recognition library, (that I can't use cause it relies on PyAudio).
For now, the Assistant's abilities are simple: It can retrieve weather or Wikipedia searches, date and time info, and jokes..
New: It can now control any media-player on the system, long as you have echo-canceling: play, pause, resume, stop, previous, next, volume up, volume down, and what's playing, all should work, including variations on ways to say those.
Thanks to Whisper, it works really well! And I should be able to add more features as I figure them out.

If anyone has any suggestions to improve how I'm doing things, I'd love to hear it!
For example, I couldn't figure out how to send numpy audio data to directly to Whisper. But I think that may require altering Whisper somehow..

Anyway, hopefully this helps other people who have the same audio-library issues as I did! ;)

**Comment by JBX2060** [comment]
Wow, just what I was looking for!!

It'll be great if you add ChatGPT's API responses next!

**Reply by UmangRajpara13** [reply]
[TalkGPT](https://github.com/UmangRajpara13/TalkGPT)

**Reply by tumor6** [reply]
Epic work. !

**Comment by palladium123** [comment]
Love this! Speech Recognition didn't work for me as intended, but this did.  With some modifications it works really well for live dictation, hardware specs notwithstanding...

**Comment by baibhavKumar1** [comment]
How to use this?
